File: seeker/snippet/two_sum.py
#date: 2025-09-08T17:11:56Z
#url: https://api.github.com/gists/4352a3c0be5a9a4e93f183bac12eac16
#owner: https://api.github.com/users/motyzk

class Solution:
    def twoSum(self, nums, target):
        complements = {}
        for i, n in enumerate(nums):
            if n in complements:
                return [complements[n], i]
            complements[target - n] = i

File: seeker/snippet/twoSum.go
//date: 2024-02-26T16:53:33Z
//url: https://api.github.com/gists/1472b977fb778c1f94dcb07140590e19
//owner: https://api.github.com/users/lbvf50mobile

// Leetcode: 1. Two Sum.
// https://leetcode.com/problems/two-sum/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 6 ms, faster than 77.05% of Go online submissions for Two Sum.
// Memory Usage: 4.2 MB, less than 19.92% of Go online submissions for Two
// Sum.
// 2024.02.26 Daily Challenge.

package main

func twoSum(nums []int, target int) []int {
	hsh := make(map[int]int)
	for i, v := range nums {
		b := target - v
		if _, ok := hsh[b]; ok {
			return []int{i, hsh[b]}
		}
		hsh[v] = i
	}
	panic("Unsolved.")
}
